experiments/experiments.go

```diff
@@ -0,0 +1,323 @@
+}
+
+// SimpleExperiment is a basic experiment choosing from a set of variants.
+type SimpleExperiment struct {
+	// id is the experiment identifier and should be unique.
+	id int
+	// name is a human-readable name of the experiment.
+	name string
+	// bucketSeed if provided, this provides the bucketSeed for determining which bucket a
+	// variant request lands in. Providing a consistent bucket bucketSeed will ensure
+	// a user is bucketed consistently. Calls to the variant method will return
+	// consistent results for any given bucketSeed.
+	bucketSeed string
+	// numBuckets determines how many available buckets there are for bucketing
+	// requests. This should match the numBuckets in the provided VariantSet.
+	// The default value is 1000, which provides a potential variant
+	// granularity of 0.1%.
+	numBuckets int
+	// enabled sets whether or not this experiment is enabled. disabling an
+	// experiment means all variant calls will return an empty string.
+	enabled bool
+	// startTime determines when this experiment is due to start. Variant
+	// requests prior to this time will return an empty string.
+	startTime time.Time
+	// endTime determines when this experiment is due to end. Variant requests
+	// after this time will return an empty string.
+	endTime time.Time
+	// logBucketing determines whether bucketing events should be logged.
+	logBucketing bool
+	// bucketVal is a string used for shifting the deterministic bucketing
+	// algorithm.  In most cases, this will be an Account's fullname.
+	bucketVal string
+	// variantSet contains a set of experimental variants as well as their
+	// distributions. It is used by experiments to track which bucket a variant
+	// is assigned to.
+	variantSet VariantSet
+	// targeting allows to target users with multiple parameters supporting
+	// both AND and OR based logical grouping.
+	targeting Targeting
+	// overrides if matched allow to force a particular variant.
+	overrides []map[string]Targeting
+}
+
+// NewSimpleExperiment returns a new instance of SimpleExperiment. Default
+// values if not otherwise provided by the ExperimentConfig will be assumed.
+func NewSimpleExperiment(experiment *ExperimentConfig) (*SimpleExperiment, error) {
+	shuffleVersion := experiment.Experiment.ShuffleVersion
+	if shuffleVersion == "" {
+		shuffleVersion = "None"
+	}
+	bucketVal := experiment.Experiment.BucketVal
+	if bucketVal == "" {
+		bucketVal = "user_id"
+	}
+	enabled := true
+	if experiment.Enabled != nil {
+		enabled = *experiment.Enabled
+	}
+	bucketSeed := experiment.Experiment.BucketSeed
+	if experiment.Experiment.BucketSeed == "" {
+		bucketSeed = fmt.Sprintf("%d.%s.%s", experiment.ID, experiment.Name, shuffleVersion)
+	}
+	variantSet, err := FromExperimentType(experiment.Type, experiment.Experiment.Variants, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	targetingConfig := experiment.Experiment.Targeting
+	if len(targetingConfig) == 0 {
+		targetingConfig = []byte(`{"OVERRIDE": true}`)
```

Comment:
I guess I'd rather separate parsing and pass in of target structure so that we can actually pass `map[string]interface{}{"OVERRIDE": true}` in here instead.